gradio/queueing.py

```diff
@@ -359,32 +349,69 @@ def get_request_params(self, websocket: fastapi.WebSocket) -> dict[str, Any]:
         return params
 
     async def call_prediction(self, events: list[Event], batch: bool):
-        data = events[0].data
-        assert data is not None, "No event data"
-        token = events[0].token
-        data.event_id = events[0]._id if not batch else None
+        body = events[0].data
+        assert body is not None, "No event data"
+        username = events[0].username
+        body.event_id = events[0]._id if not batch else None
         try:
-            data.request = self.get_request_params(events[0].websocket)
+            body.request = self.get_request_params(events[0].websocket)
         except ValueError:
             pass
 
         if batch:
-            data.data = list(zip(*[event.data.data for event in events if event.data]))
-            data.request = [
+            body.data = list(zip(*[event.data.data for event in events if event.data]))
+            body.request = [
                 self.get_request_params(event.websocket)
                 for event in events
                 if event.data
             ]
-            data.batched = True
-        response = await AsyncRequest(
-            method=AsyncRequest.Method.POST,
-            url=f"{self.server_path}api/predict",
-            json=dict(data),
-            headers={"Authorization": f"Bearer {self.access_token}"},
-            cookies={"access-token": token} if token is not None else None,
-            client=self.queue_client,
+            body.batched = True
+
+        app = self.server_app
+        if app is None:
+            raise Exception("Server app has not been set.")
+        api_name = "predict"
+
+        fn_index_inferred = route_utils.infer_fn_index(
+            app=app, api_name=api_name, body=body
+        )
+
+        gr_request = route_utils.compile_gr_request(
```

Comment:
Thank you! You are right :D
* I moved `restore_session_state` into `call_process_api`.
* I'm still wondering if I should move `traceback.print_exc()` into it, because it's more interface-like code printing output, but not a logic, so putting it on the current position also makes sense, for example `predict()` in `routes.py` that is an endpoint handler or the I/O layer.